interfaces.md

```diff
@@ -0,0 +1,73 @@
+# Interfaces
+
+Document that outlines what each layer requires and provides in more detail.
```

Comment:
"Detailed document outlining what each layer requires and provides" reads better